hpct-ldap-server/src/servers/ldap.py

```diff
@@ -0,0 +1,503 @@
+
+            if rc != 0:
+                raise Exception("Unable to create server certificate.")
+
+            # Adjust Permissions and Ownership
+            g_args = ["chgrp", "openldap", "/etc/ldap/ldap01_slapd_key.pem"]
+            rc = subprocess.call(g_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+            if rc != 0:
+                raise Exception("Unable to set group permissions.")
+
+            o_args = ["chmod", "0640", "/etc/ldap/ldap01_slapd_key.pem"]
+            rc = subprocess.call(o_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+            if rc != 0:
+                raise Exception("Unable to set owner permissions.")
+
+            # Server Certificates LDIF
+            sc_ldif = [
+                "dn: cn=config",
+                "add: olcTLSCACertificateFile",
+                "olcTLSCACertificateFile: /etc/ssl/certs/mycacert.pem",
+                "-",
+                "add: olcTLSCertificateFile",
+                "olcTLSCertificateFile: /etc/ldap/ldap01_slapd_cert.pem",
+                "-",
+                "add: olcTLSCertificateKeyFile",
+                "olcTLSCertificateKeyFile: /etc/ldap/ldap01_slapd_key.pem",
+            ]
+            with open("/etc/ldap/certinfo.ldif", "w") as f:
+                for line in sc_ldif:
+                    f.write(f"{line}\n")
+
+            # Apply Certificate LDIF
+            ldif_args = [
+                "ldapmodify",
+                "-Y",
+                "EXTERNAL",
+                "-H",
+                "ldapi:///",
+                "-f",
+                "/etc/ldap/certinfo.ldif",
+            ]
+            rc = subprocess.call(ldif_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+            if rc != 0:
+                raise Exception("Unable to apply Server Certificate LDIF")
+        else:
+            raise Exception("Organization not set.")
+
+    def tls_load(self, domain=None):
+        """Load and return CA cert and ldap uri.
+
+        Parameters
+        ----------
+        domain : str
+            Domain name.
+        """
+        if domain:
+            dcs = self._split_domain(domain)
+
+            # sssd conf template for clients
+            ldap_uri = subprocess.run(["cat", "/etc/hostname"], capture_output=True, text=True)
+            sssd_conf = [
+                "[sssd]",
+                "config_file_version = 2",
+                f"domains = {domain}",
+                "",
+                f"[domain/{domain}]",
+                "id_provider = ldap",
+                "auth_provider = ldap",
+                f"ldap_uri = ldap://{ldap_uri.stdout}",
+                "cache_credentials = True",
+                f"ldap_search_base = {dcs}",
+            ]
+            with open("/etc/sssd/sssd_conf.template", "w") as f:
+                for line in sssd_conf:
+                    f.write(f"{line}\n")
```

Comment:
As above, this could be `"\n".join(sssd_conf)`. Or, the `sss_conf` could just be a strings with newline at the end of each. Then no need to concatenate them:
```
ssd_conf = (
    "[sssd]\n"
    "config_file_version = 2\n"
    ...
)
```